### DISCUSSION TITLE
Imports only used on server side (like axios) are bundled together?

### DISCUSSION THREAD

**Discussion by kimgiutae** [vercel/next.js#27681] [Help] [answered]
I use axios for fetching data from my CMS.
Of course I have to import the library on all my pages.
But I only use it on getStaticProps or getStaticPaths.

I wonder, cause the import is not dynamic, is that scripts sended to the user?
Or next recognize that is only used on server side and is not bundled.

**Comment by zeekrey** [comment, marked as answer]
You can check the actual client code by viewing the source or check it here: https://next-code-elimination.vercel.app/ If a dependency is only used within getStaticProps, it won't be send to the client for sure.

Dynamic imports have nothing todo with this technique imo: https://nextjs.org/docs/advanced-features/dynamic-import

**Reply by kimgiutae** [reply]
Hi @zeekrey. The cliente code is full of code, and the js are splited in various chunks... I really don't know were to look at.

**Reply by zeekrey** [reply]
Then paste your code on the left side of this page and check the imports: https://next-code-elimination.vercel.app/

**Reply by kimgiutae** [reply]
Oh. I see now. Very useful tool https://next-code-elimination.vercel.app/. Thanks
Y dunno why is sending the types imports but at least axios is not sended to the client.
